### ISSUE TITLE
[Woo POS] Multiple readers found can be shown empty

### ISSUE THREAD

**Issue by joshheald** [woocommerce/woocommerce-ios#13778]
> https://github.com/user-attachments/assets/1e5c0721-8796-40f2-b01e-a44c7234e5a8
> 
> Note that this video shows an issue at the end where the Multiple Readers View is shown empty after tapping try again on the error. That's an underlying bug, and out of the scope to fix here, but I'll raise an issue.
> Found in #13775 

The multiple readers alert should not show empty... unless it becomes empty while on screen.

If it's showing, it should continue searching.

In this case, it's probably ignoring the two previously found readers.

## Repro
Use the simulated card reader, and set `Terminal.shared.simulatorConfiguration.availableReaderUpdate = .lowBattery` in the `StripeCardReaderService`

1. Launch the app and open POS
2. Tap `Connect your reader`
3. Connect to the first discovered reader
4. When the update ends in a low battery error, tap cancel
5. Observe that the `multiple readers found` screen is shown empty, and no readers are ever shown.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by malinajirka** [comment]
@joshheald Is this still valid and worth spending time on?

**Comment by joshheald** [comment]
Yes it's still valid, but it's not common so I don't think we should spend much time on it. 

It looks like there's a fairly easy fix which will also benefit IPP. I'll try to do it in a timebox this week.

**Comment by joshheald** [comment]
This happens because we set the state to `.searching` in the cancel block from an update error, without having reset the connection controller. 

This appears to be deliberate, from the tests, and makes sense to enable the merchant to try connecting again when they've plugged it in (though it shouldn't really be labelled cancel or close.)

However, the `StripeCardReaderService` resets its internal state when we get this error, so the previously found readers aren't shown any more.

A better approach would be to have the retry button set the state to `.retry`, and then have a `Cancel/Close` button as well.
